accounts/abi/bind/backends/simulated.go

```diff
@@ -607,16 +607,14 @@ func (b *SimulatedBackend) callContract(ctx context.Context, call ethereum.CallM
 	// Execute the call.
 	msg := callMsg{call}
 
-	txContext := core.NewEVMTxContext(msg)
-	evmContext := core.NewEVMBlockContext(block.Header(), b.blockchain, nil)
 	// Create a new environment which holds all relevant information
 	// about the transaction and calling mechanisms.
+	txContext := core.NewEVMTxContext(msg)
+	evmContext := core.NewEVMBlockContext(block.Header(), b.blockchain, nil)
 	vmEnv := vm.NewEVM(evmContext, txContext, stateDB, b.config, vm.Config{NoBaseFee: true})
 	gasPool := new(core.GasPool).AddGas(math.MaxUint64)
-
 	vmRunner := b.blockchain.NewEVMRunner(block.Header(), stateDB)
-
-	return core.NewStateTransition(vmEnv, msg, gasPool, vmRunner).TransitionDb()
+	return core.NewStateTransition(vmEnv, msg, gasPool, vmRunner, nil).TransitionDb()
```

Comment:
You've set `sysctx` to be nil several times. It's fine if it's not initialized pre-espresso, but not initializing it post-espresso could cause some problems because `ApplyTransaction` expects it to be well formed post-espresso.